csrc/python_frontend/fusion_definition.cpp

```diff
@@ -87,13 +87,19 @@ void FusionDefinition::finalizeDefinition() {
     }
     trie_node_ = fusionCache()->createChild(trie_node_, end_record_.get());
     fusion_id_ = std::optional<size_t>(trie_node_->fusion_id);
-    NVF_CHECK(id().has_value(), "Invalid fusion id!");
+    try {
+      NVF_CHECK(id().has_value(), "Invalid fusion id!");
 
-    if (isDebugDumpEnabled(DebugDumpOption::PythonDefinition)) {
-      print(debug());
-    }
+      if (isDebugDumpEnabled(DebugDumpOption::PythonDefinition)) {
+        print(debug());
+      }
 
-    buildFusionIr(preschedFusion());
+      buildFusionIr(preschedFusion());
+    } catch (const std::exception& e) {
+      trie_node_->setException(e.what());
```

Comment:
issue happening after `fusionCache()->createChild` wouldn't be visible by fusion cache, if the exception is suppressed by python side.

Hence we explicitly set the exception message on the terminal node, so we'll be able to throw the same exception again when user tries to create the same fusion entry.

Reply:
Can you put something to that effect in a comment at this location?